test/test_torch.py

```diff
@@ -14639,6 +14639,40 @@ def test_ctor_with_numpy_array(self, device):
             for i in range(len(array)):
                 self.assertEqual(tensor[i], array[i])
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_numpy_scalar_cmp(self, device, dtype):
+        t = torch.tensor([0, 1, 2], dtype=dtype, device=device)
+        if dtype == torch.bfloat16:
+            with self.assertRaises(TypeError):
+                np_array = t.cpu().numpy()
+            return
+
+        np_array = t.cpu().numpy()
+        self.assertEqual(t[0], np_array[0])
```

Comment:
Currently the complex tensors won't actually have any complex values and there's only a single 1D tensor. What about extending the test slightly to support complex values and multiple tensor sizes? It could do something like this:

```
is_complex = dtype in (torch.cfloat, torch.cdouble)
if is_complex:
  tensors = (...)
else:
  tensors = (
    torch.tensor(5, dtype=dtype, device=device),
    torch.tensor([0, 1, 2], dtype=dtype, device=device),
    torch.tensor([[1, 2], [3, 4]], dtype=dtype, device=device),
  )

for tensor in tensors:
  np_array = tensor.cpu().numpy()
  for t, a in product((tensor[0], tensor[0].item()), (np_array[0], np_array[0].item()):
    self.assertEqual(t, a)
    if not is_complex:
      self.assertTrue(t == a)

  if is_complex:
    self.assertTrue(t[0] == np_array[0].item())
    self.assertTrue(t[0].item() == np_array[0])
    self.assertFalse(t[0] == np_array[0])  
```

Reply:
Sounds good

Reply:
Per the following discussion, if you follow this pattern the later conditionals will probably need to check `dtype is torch.cfloat` instead of the `is_complex` boolean.

Reply:
What do you mean?

Reply:
Won't `self.assertFalse(t[0] == np_array[0])` be true only for cfloat, not for cdouble?

Reply:
Yes, thanks. Anyway, I have structured the test a bit differently.